index.bs

```diff
@@ -201,11 +373,65 @@ To <dfn for=XRRay>obtain the matrix</dfn> for a given {{XRRay}} |ray|
     </dl>
   1. Let |translation| be the translation matrix with components corresponding to |ray|'s {{XRRay/origin}}
   1. Let |matrix| be the result of premultiplying |rotation| from the left onto |translation| (i.e. <code>translation * rotation</code>) in column-vector notation.
-  1. Set |ray|'s [=XRRay/internal matrix=] to |matrix|
-  1. Return |matrix|
+  1. Set |ray|'s [=XRRay/matrix=] to |matrix|.
+  1. Return |matrix|.
 
 </div>
 
+<div class="algorithm" data-algorithm="distance-along-ray">
+
+The <dfn>distance along the ray</dfn>, |distance|, from {{XRRay}} |ray| to any entity |entity| is defined such that <code>|ray|.origin + |ray|.direction * |distance|</code> results in a point beloning to the entity |entity|, |distance| is non-negative, and there does not exist a smaller value of |distance| for the above predicate to still hold. It is up to the [=/XR device=] to define the meaning of "point belonging to an entity".
+
+</div>
+
+Native device concepts {#native-device-concepts} 
+======================
+
+<section class="non-normative">
+
+User agents implementing hit test API must have a way of obtaining information about user's environment from underlying XR device. This section attempts to describe requirements and concepts related to native capabilities of the device and is by neccesity sufficiently under-specified to leave ample room for different underlying frameworks / devices.
+
+</section>
+
+Native hit test {#native-hit-test-section}
+---------------
+
+In this specification it is assumed that [=/XR device=] exposes a way for the user agent to perform a <dfn>native hit test</dfn> that satisfies the following requirements:
+- Accepts a 3D ray that will be tested against user's environment.
+- Returns a collection of 3D poses representing intersection points of the passed in ray with user's environment. Each entry in the collection should also contain information about the type of the native entity that was used to obtain that native result and enough information to enable the user agent to compute surface normal to the user's environment at the intersection point.
+
+Note: For devices that do not expose the hit test functionality natively, it might still be possible for user agents to implement this specification by leveraging other ways of obtaining the information about user's environment that might be exposed by the XR device.
```

Comment:
Is the surface normal of the plane returned somewhere?

Reply:
Indirectly - it will be equivalent to the y axis of the XRPose returned from XRHitTestResult.getPose(...). Let me state it explicitly in the text, I don't think I mention it anywhere.

Reply:
ah, that makes sense.  Yes, please make that explicit!

Reply:
I've added some more explicit phrasing around this (both in hit test result section and in native hit test result section) - let me know if you think this is sufficient. We should also decide whether spec should be even more precise when describing the coordinate system defined by hit test result's native origin - I have added it as an inline issue in the draft.